### PR TITLE
feat(ATS): add vuid manager

### PR DESCRIPTION
**Pull request by jaeopt**
## Summary
Add VuidManager which will create and manage a vuid:
 
- create a random vuid if it cannot be found in the device with the required format.
- "vuid_" + 32 hexadecimal string (UUID with all "-"s removed). 
- all lowercases ("vuid_330e05cad15746d9af8a75b8d1050a87")
- SDK will reuse the same vuid (saving to and retrieving from a persistent storage when the client application starts).
- SDK will support a single vuid shared by multiple SDK instances (thread-safe)
- ODP module added

## Test plan
- add unit tests

## Issues
- [FSSDK-8548](https://optimizely.atlassian.net/browse/FSSDK-8548)

### CODE REVIEW

File: odp/src/androidTest/java/com/optimizely/ab/android/odp/VuidManagerTest.kt [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        // remove vuid storage
+        cleanSheredPrefs()
+        // remove a singlton instance
+        VuidManager.removeSharedForTesting()
+
+        vuidManager = VuidManager.getShared(context)
+    }
+
+    @After
+    fun cleanSheredPrefs() {
```
Comment (opti-jnguyen) [inline odp/src/androidTest/java/com/optimizely/ab/android/odp/VuidManagerTest.kt:43]: Should this be `cleanSharedPrefs`?
Typo in "Shered" to "Shared".

File: event-handler/src/androidTest/java/com/optimizely/ab/android/event_handler/EventDAOTest.java [hunk 2]
```diff
@@ -54,11 +54,12 @@ public void setupEventDAO() {
         logger = mock(Logger.class);
         context = InstrumentationRegistry.getInstrumentation().getTargetContext();
         eventDAO = EventDAO.getInstance(context, "1", logger);
+        tearDownEventDAO();
     }
 
     @After
     public void tearDownEventDAO() {
-        assertTrue(context.deleteDatabase(String.format(EventSQLiteOpenHelper.DB_NAME , "1")));
+        context.deleteDatabase(String.format(EventSQLiteOpenHelper.DB_NAME , "1"));
```
Comment (opti-jnguyen) [inline event-handler/src/androidTest/java/com/optimizely/ab/android/event_handler/EventDAOTest.java:62]: Out of curiosity, why was the assertion removed here?

File: test-app/src/main/java/com/optimizely/ab/android/test_app/NotificationService.kt [hunk 3]
```diff
@@ -20,7 +20,8 @@ import android.content.Intent
 import androidx.core.app.NotificationCompat
 import com.optimizely.ab.android.shared.CountingIdlingResourceManager
 
-class NotificationService : IntentService("NotificationService") {
+class
```
Comment (opti-jnguyen) [inline test-app/src/main/java/com/optimizely/ab/android/test_app/NotificationService.kt:23]: Is `class` intentionally on a separate line here?

**Review by opti-jnguyen — CHANGES_REQUESTED** [review]
Overall LGTM besides the probable typo with `shered`.

Food for thought - what do you think about abstracting out VUID as its own class, separate from VUIDManager?

Was fun re-visiting Kotlin!

File: event-handler/src/androidTest/java/com/optimizely/ab/android/event_handler/EventDAOTest.java (hunk 2 above)
Comment (jaeopt) [reply event-handler/src/androidTest/java/com/optimizely/ab/android/event_handler/EventDAOTest.java:63]: For cleaning up with remaining database from other tests, the teardown is also called when starting each test. So assert is not true always.

File: test-app/src/main/java/com/optimizely/ab/android/test_app/NotificationService.kt (hunk 3 above)
Comment (jaeopt) [reply test-app/src/main/java/com/optimizely/ab/android/test_app/NotificationService.kt]: Not intentional. restored :)

**Review by opti-jnguyen — APPROVED** [review]
LGTM

**Pull request merged**